macros/src/test/scala/code/chymyst/jc/MacrosSpec.scala

```diff
@@ -63,6 +69,28 @@ class MacrosSpec extends FlatSpec with Matchers with BeforeAndAfterEach {
 
   behavior of "macros for inspecting a reaction body"
 
+  it should "fail to compile a reaction with empty singleton clause" in {
```

Comment:
Nice work by scalatest matchers (shouldNot compile and various others of the sort)!